staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -975,3 +985,14 @@ func (az *Cloud) ShouldNodeExcludedFromLoadBalancer(nodeName string) (bool, erro
 
 	return az.excludeLoadBalancerNodes.Has(nodeName), nil
 }
+
+// This, along with the few lines that call this function in updateNodeCaches, should be
+// replaced by https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1195 once that merges.
+func isNodeReady(node *v1.Node) bool {
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady && cond.Status == v1.ConditionTrue {
```

Comment:
Thanks for the PR. Could we also check whether this is a new node or not here (via "node.cloudprovider.kubernetes.io/uninitialized")? new node should not be removed so as to reduce number of ARM calls. 

https://github.com/kubernetes/cloud-provider/blob/5a236942a3ca6f0454bfcb979cc31207af4894b2/api/well_known_taints.go#L20-L24

Reply:
Sure! I just modified the condition on lines 826 and 865. Let me know what you think. Thanks!

Reply:
Sure! I just modified the condition on lines 826 and 865. Let me know what you think. Thanks!

Reply:
Reducing the number of ARM calls (don't know what ARM is) is a worthy goal as long as the feature for an end user still works. This seems to make that feature not work: a node that has not been initialized by the cloud provider will be unable to participate in service routing, so adding it to the load balancer will break for users trying to use the service type load balancer when this uninitialized node is added to the the LB.

Reply:
It would not impact services because LB probing would fail.

As discussed in https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1195#issuecomment-1050894175, we should not exclude new Nodes from LB backend pool, but there is no taint if the cluster is not using CCM. Any ideas to check whether a Node is new or not?